wicker/core/datasets.py

```diff
@@ -207,6 +271,106 @@ def schema(self) -> DatasetSchema:
                 )
         return self._schema
 
+
+class FileSystemDataset(BaseDataset):
+    """Implementation of a Map-based dataset on local file system or mounted drive"""
+
```

Comment:
IMO even though `FileSystemDataset` is now basically just `BaseDataset`, it is still valuable to declare a subclass and `__init__` function because the `storage` and `pa_filesystem` objects now have to have a more specific subclass type. This will help prevent any errors when creating instances of this class.